### DISCUSSION TITLE
Unable to use IN operator in 2.0.36

### DISCUSSION THREAD

**Discussion by Apakottur** [sqlalchemy/sqlalchemy#12018] [Usage Questions]
The following code:
```python
result = await session.execute(
            select(Actor.id).where(
                Actor.id.in_([Actor.id]),
            )
````

Is valid in sqlalchemy==2.0.35 and generates the following correct SQL:
```sql
SELECT actor.id  FROM actor WHERE actor.id IN (actor.id)
```

However, in sqlalchemy==2.0.36 the same expression gives:
```text
sqlalchemy.exc.ArgumentError: IN expression list, SELECT construct, or bound parameter object expected, got [<sqlalchemy.orm.attributes.InstrumentedAttribute object at 0x7b8fec7bb240>].
```

It this a bug or using IN like this is no longer allowed?

[full gist](https://gist.github.com/Apakottur/3ae21382dcd22539a86464820269b613)

**Comment by CaselIT** [comment]
Hi,

Ignoring the fact that such an IN does not make much sense as written, this seems to be a regression caused by https://github.com/sqlalchemy/sqlalchemy/commit/c1bffa312ba973aaac639e04950457e81d291f9b
@zzzeek I think this is a valid use case for which we do not seem to have any test

**Reply by CaselIT** [reply]
please follow on https://github.com/sqlalchemy/sqlalchemy/issues/12019

**Reply by Apakottur** [reply]
Thanks, I tried to create a minimal reproducible example so the query is a bit pointless.

If it matters, my real use case checks if one column is equal to one of two others:

```sql
SELECT * FROM actor WHERE actor.field_1 IN (actor.field_2, actor.field_3)
```

**Reply by CaselIT** [reply]
I assumed as much, and I suggested a similar use case in a comment in the issue
